src/index.ts

```diff
@@ -6,27 +6,76 @@ import { IThemeManager } from '@jupyterlab/apputils';
 import { themeEditorIcon } from './icons';
 import { ThemeEditorModel } from './model';
 import { ThemeEditorView } from './view';
-import { IChangedArgs } from '@jupyterlab/coreutils';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+import { ITranslator } from '@jupyterlab/translation';
+
+const PLUGIN_ID = 'jupyter-theme-editor:plugin';
 
 /**
  * Initialization data for the jupyter-theme-editor extension.
  */
 const plugin: JupyterFrontEndPlugin<void> = {
-  id: 'jupyter-theme-editor:plugin',
+  id: PLUGIN_ID,
   autoStart: true,
   requires: [IThemeManager],
-  activate: async (app: JupyterFrontEnd, themeManager: IThemeManager) => {
-    const onThemeChanged = (
-      themeManager: IThemeManager,
-      changes: IChangedArgs<string, string | null, string>
-    ) => {
+  optional: [ISettingRegistry, ITranslator],
+  activate: (
+    app: JupyterFrontEnd,
+    themeManager: IThemeManager,
+    settingRegistry: ISettingRegistry,
+    translator: ITranslator
+  ) => {
+    const onThemeChanged = (themeManager: IThemeManager) => {
       themeManager.themeChanged.disconnect(onThemeChanged);
       const model = new ThemeEditorModel();
-      const view = new ThemeEditorView(model);
+      const view = new ThemeEditorView(model, translator);
       view.addClass('jp-theme-editor-view-panel');
       view.id = 'theme-editor';
       view.title.icon = themeEditorIcon;
+      model.schema = {};
       app.shell.add(view, 'left');
+      console.log('JupyterLab extension jupyter-theme-editor is activated!');
+
+      Promise.all([app.restored, settingRegistry.load(PLUGIN_ID)]).then(
+        ([, setting]) => {
+          model.schema = setting.schema.properties?.customStyles;
```

Comment:
You can protect this by adding a fallback value if `properties` is undefined:

```suggestion
          model.schema = setting.schema.properties?.customStyles ?? {};
```